Jules.md

```diff
@@ -0,0 +1,553 @@
+    using GCC 5+, define `-D_GLIBCXX_USE_CXX11_ABI=0`.
+
+## Including the SDK in Projects
+
+### CMake Projects
+
+Use `add_subdirectory()` in your `CMakeLists.txt`:
+
+```cmake
+add_subdirectory("[[Path to the Firebase C++ SDK]]")
+target_link_libraries([[Your CMake Target]] firebase_analytics firebase_app)
+```
+
+### Android Gradle Projects
+
+In addition to CMake setup, use `Android/firebase_dependencies.gradle` in your
+`build.gradle`:
+
+```gradle
+apply from: "[[Path to the Firebase C++ SDK]]/Android/firebase_dependencies.gradle"
+firebaseCpp.dependencies {
+  analytics
+}
+```
+
+For more detailed instructions and examples, always refer to the main
+`README.md` and the
+[C++ Quickstarts](https://github.com/firebase/quickstart-cpp).
+
+# Testing
+
+## Testing Strategy
+
+The primary method for testing in this repository is through **integration
+tests** for each Firebase library. While the `README.md` mentions unit tests
+run via CTest, the current and preferred approach is to ensure comprehensive
+coverage within the integration tests.
+
+## Running Tests
+
+*   **Integration Test Location**: Integration tests for each Firebase product
+    (e.g., Firestore, Auth) are typically located in the `integration_test/`
+    directory within that product's module (e.g.,
+    `firestore/integration_test/`).
+*   **Test Scripts**: The root of the repository contains scripts for running
+    tests on various platforms, such as:
+    *   `test_windows_x32.bat`
+    *   `test_windows_x64.bat`
+    *   `test_linux.sh`
+    *   `test_mac_x64.sh`
+    *   `test_mac_ios.sh`
+    *   `test_mac_ios_simulator.sh`
+
+    These scripts typically build the SDKs and then execute the relevant tests
+    (primarily integration tests) via CTest or other platform-specific test
+    runners.
+
+## Writing Tests
+
+When adding new features or fixing bugs:
+
+*   Prioritize adding or updating integration tests within the respective
+    product's `integration_test/` directory.
+*   Ensure tests cover the new functionality thoroughly and verify interactions
+    with the Firebase backend or other relevant components.
+*   Follow existing patterns within the integration tests for consistency.
+
+# API Surface
+
+## General API Structure
+
+The Firebase C++ SDK exposes its functionality through a set of classes and
+functions organized by product (e.g., Firestore, Authentication, Realtime
+Database).
+
+### Initialization
+
+1.  **`firebase::App`**: This is the central entry point for the SDK.
+    *   It must be initialized first using `firebase::App::Create(...)`.
+    *   On Android, this requires passing the JNI environment (`JNIEnv*`) and
+        the Android Activity (`jobject`).
+    *   `firebase::AppOptions` can be used to configure the app with specific
+        parameters if not relying on a `google-services.json` or
+        `GoogleService-Info.plist` file.
+2.  **Service Instances**: Once `firebase::App` is initialized, you obtain
```

Comment:
Do you want to add a thing that some products (like Analytics) are singletons that initialize differently?